src/modules/plymouthcfg/main.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# === This file is part of Calamares - <http://github.com/calamares> ===
+#
+#   Copyright 2016, Artoo <artoo@manjaro.org>
+#
+#   Calamares is free software: you can redistribute it and/or modify
+#   it under the terms of the GNU General Public License as published by
+#   the Free Software Foundation, either version 3 of the License, or
+#   (at your option) any later version.
+#
+#   Calamares is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#   GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License
+#   along with Calamares. If not, see <http://www.gnu.org/licenses/>.
+
+import libcalamares
+
+from libcalamares.utils import debug, target_env_call
+
+
+class PlymouthController:
+
+    def __init__(self):
+        self.__root = libcalamares.globalstorage.value('rootMountPoint')
+
+    @property
+    def root(self):
+        return self.__root
+
+    def setTheme(self):
+        plymouth_theme = libcalamares.job.configuration["plymouth_theme"]
+        target_env_call(["sed", "-e", 's|^.*Theme=.*|Theme=' +
+                         plymouth_theme + '|', "-i", "/etc/plymouth/plymouthd.conf"])
+
+    def detect(self):
+        plymouth_bin = target_env_call(["which", "plymouth"])
```

Comment:
I don't understand the point of using "`sh -c`". If `target_env_call` delivers the exit code, it is redundant. If `target_env_call` does not deliver the exit code, it won't help, because then it won't deliver the exit code of `sh` any more than the one of `which`. So in that case, it is also redundant.